code-the-dream-react/src/AddTodoForm.js

```diff
@@ -1,18 +1,27 @@
 import React from 'react';
 
-function AddTodoForm(props) {
+function AddTodoForm({ onAddTodo }) {
+    const [todoTitle, setTodoTitle] = React.useState(''); // setTodoTitle in this case is sometimes called the function updater/callback handler
+
+    function handleTitleChange(event) {
+        let newTodoTitle = event.target.value;
+        setTodoTitle(newTodoTitle);
+    };
+
     function handleAddTodo(event) {
         event.preventDefault();
-        let todoTitle = event.target.title.value;
-        console.log(todoTitle);
-        document.getElementById("formValue").reset();
-        props.onAddTodo(todoTitle);   /* I am not 100% sure if this is where I was expected to add the onAddTodo callback prop within the handleAddTodo function */                 
+        let todoItem = {
+            title: todoTitle,
+            id: Date.now(),
+            }
+        onAddTodo(todoItem);
```

Comment:
do we want to be able to call the `onAddTodo()` even if the title is an empty string?

how could we handle this scenario?

ps. no need to make changes but let's start thinking about those edge cases

Reply:
Ah, ok. My guess is that I would need a conditional statement to take care of the edge cases, but I was just following the steps and it must have crossed my mind! Thank you, Orlando!